src/classes/AFFL_AccChange_TDTM.cls

```diff
@@ -132,17 +149,63 @@ public with sharing class AFFL_AccChange_TDTM extends TDTM_Runnable {
                    	i++;
                     //Turn off CON_PrimaryAffls_TDTM_After_Update temporarily to avoid creating duplicate affiliation
                     TDTM_ProcessControl.turnOffRecursionFlag(TDTM_ProcessControl.registeredTrigger.CON_PrimaryAffls_TDTM_After_Update);
+
                	}
+                if (afflsMadePrimary.size() > 0) {
+                    afflMulti.uncheckOtherPrimariesSameType(afflsMadePrimary, dmlWrapper);
+                }
            	}
         }
 
         TDTM_TriggerHandler.processDML(dmlWrapper, true);
         dmlWrapper = null;  
-        if(triggerAction == TDTM_Runnable.Action.AfterUpdate)
-        {      
+        if (triggerAction == TDTM_Runnable.Action.AfterUpdate) {      
           	TDTM_ProcessControl.setRecursionFlag(TDTM_ProcessControl.registeredTrigger.AFFL_AccChange_TDTM, false);
             TDTM_ProcessControl.resetRecursionFlag(TDTM_ProcessControl.registeredTrigger.CON_PrimaryAffls_TDTM_After_Update);
         }
         return dmlWrapper;
     }
+   
+    /******************************************************************************************************************************
+    * @description creates a map Affiliation Id to other Primary Affilaitons with same Account and Contact
+    * @param newAffiliationList List of affiliations that are being updated
+    * @return idToOtherAffiliationWithSameAccAndCon A map Affiliation Id to other Primary Affilaitons with same Account and Contact
+    ********************************************************************************************************************************/
+    @testVisible
+    private Map<Id, List<Affiliation__c>> createdIdToOtherPrimaryAfflsWithSameAccAndCon(List<Affiliation__c> newAffiliationList) {
+
+        Set<ID> newPrimaryAffls = new Set<ID>();
+        Set<ID> afflContactIDs = new Set<ID>();
+        Set<String> afflAccIds = new Set<String>();
+        List<Affiliation__c> oldPrimaryAffls = new List<Affiliation__c>();
+
+        for (Affiliation__c affl : newAffiliationList) {
+            newPrimaryAffls.add(affl.ID);
+            afflContactIDs.add(affl.Contact__c);
+            afflAccIds.add(affl.Account__c);
+        }
+        
+        // Get all possible Primary Affiliation with Same Contact and Account
+        List<Affiliation__c> otherPrimaryAffls = [SELECT Id, Contact__c, Account__c, Affiliation_Type__c, Primary__c
+                                                  FROM Affiliation__c WHERE Account__c IN :afflAccIds and Contact__c IN :afflContactIDs
+                                                  AND Primary__c = TRUE AND Id NOT IN :newPrimaryAffls];
+        
+        Map<Id, List<Affiliation__c>> idToOtherAffiliationWithSameAccAndCon = new Map<Id, List<Affiliation__c>>();
+
+        // Organize the results in a map
+        for (Affiliation__c otherPrimaryAffl : otherPrimaryAffls) {
```

Comment:
Please restructure this code. The nested for loop iterates through non-relevant records.

Reply:
@MatthewBlanski Thanks for the quick review, I have addressed all the feedback. I didn't use any value object but reduced number of iterations. Can you please give it another shot? Thx.